compiler/damlc/tests/daml-test-files/List.daml

```diff
@@ -115,7 +111,7 @@ testDedupSort = scenario do
     assert $ m == [1,2,3,4,5,7]
 
 testDedupBy = scenario do
-  [2, 1, 3] === dedupBy compare [2, 1, 3, 2, 1]
+  [2, 1, 3] === dedup [2, 1, 3, 2, 1]
```

Comment:
The test even has `dedupBy` in the name so we should use it.

Reply:
Yes, but `dedup = dedupBy compare` hence it is being used.

Reply:
we have separate tests for `dedup` above. `dedup = dedupBy compare` is how we implemented it now but if we change that this test should still test that `dedupBy compare` works properly.

Reply:
fine